In [ ]:
import pandas as pd

data = pd.read_csv("data_kantin.csv")

In [ ]:
import pandas as pd

data = pd.read_csv("data_kantin.csv")

In [ ]:
import pandas as pd

data = pd.read_csv("data_kantin.csv")

print(data)

        tanggal         menu kategori  harga  terjual
0    2026-08-20   Teh Hangat  Minuman   3000     11.0
1    2026-08-31  Nasi Goreng  Makanan  12000     29.0
2    2026-08-21  Nasi Goreng  Makanan  12000     28.0
3    2026-08-28   Teh Hangat  Minuman   3000     12.0
4    2026-08-21     Es Jeruk  Minuman   5000     19.0
..          ...          ...      ...    ...      ...
105  2026-09-01     Es Jeruk  Minuman   5000     24.0
106  2026-09-04  Nasi Goreng  Makanan  12000     28.0
107  2026-08-19      Kerupuk    Snack   2000     37.0
108  2026-08-25       Es Teh  Minuman   4000     44.0
109  2026-09-02      Kerupuk    Snack   2000     34.0

[110 rows x 5 columns]


In [ ]:
import numpy as np

LATIHAN 1

In [ ]:
import numpy as np
harga = np.array([5000, 7000, 3000, 12000, 4500])
print('Rata-rata harga:', harga.mean())
print('Harga tertinggi:', harga.max())
print('Harga setelah diskon 10%:', harga * 0.9)

Rata-rata harga: 6300.0
Harga tertinggi: 12000
Harga setelah diskon 10%: [ 4500.  6300.  2700. 10800.  4050.]


ANALISIS LATIHAN 1

harga = np.array([5000, 7000, 3000])

print(harga * 0.9)

In [ ]:
harga = np.array([5000, 7000, 3000])

print(harga * 0.9)

[4500. 6300. 2700.]


LATIHAN 2

In [ ]:
import pandas as pd
data_kantin = {
'menu': ['Nasi Goreng', 'Es Teh', 'Mie Ayam', 'Es Teh', None],
'harga': [12000, 4000, 10000, 4000, 8000],
'terjual': [23, 40, None, 35, 18]
}
df = pd.DataFrame(data_kantin)
print(df)

          menu  harga  terjual
0  Nasi Goreng  12000     23.0
1       Es Teh   4000     40.0
2     Mie Ayam  10000      NaN
3       Es Teh   4000     35.0
4         None   8000     18.0


ANALISIS LATIHAN 2

jika data set ada yang kosong dan tidak dibersihkan, maka hasil analisis tidak akan akurat

Latihan 3

In [ ]:
import pandas as pd

data = pd.read_csv("data_kantin.csv")

df = pd.read_csv('data_kantin.csv')
print(df.head()) # 5 baris pertama
print(df.info()) # tipe data & jumlah non-null tiap kolom
print(df.describe()) # statistik ringkas kolom numerik
print(df.shape) # jumlah (baris, kolom)

      tanggal         menu kategori  harga  terjual
0  2026-08-20   Teh Hangat  Minuman   3000     11.0
1  2026-08-31  Nasi Goreng  Makanan  12000     29.0
2  2026-08-21  Nasi Goreng  Makanan  12000     28.0
3  2026-08-28   Teh Hangat  Minuman   3000     12.0
4  2026-08-21     Es Jeruk  Minuman   5000     19.0
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 110 entries, 0 to 109
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   tanggal   110 non-null    object 
 1   menu      107 non-null    object 
 2   kategori  110 non-null    object 
 3   harga     110 non-null    int64  
 4   terjual   104 non-null    float64
dtypes: float64(1), int64(1), object(3)
memory usage: 4.4+ KB
None
              harga     terjual
count    110.000000  104.000000
mean    6336.363636   22.923077
std     3520.001232   10.831178
min     2000.000000    8.000000
25%     3000.000000   13.000000
50%     5000.000000   20.000000
75%    10000.000000 

# kolom menu dan juga terjual, artinya mereka masing masing memiliki beberapa kode yang hilang / kosong

Latihan 4


In [ ]:
print(df.isnull().sum()) # jumlah data kosong tiap kolom
df['terjual'] = df['terjual'].fillna(0) # isi kekosongan dengan 0
df = df.dropna(subset=['menu']) # hapus baris jika kolom menu kosong

tanggal     0
menu        3
kategori    0
harga       0
terjual     6
dtype: int64


Analisis Latihan 4

karna di kolom jual lebih logis di isi 0 karna itu sebagai laporan penjualan, sedangkan untuk menu kosong yang tidak bisa di isi 0 ataupun menebak nebak karna itu sebuah nama, jadi dihapus

In [ ]:
print(df.duplicated().sum()) # jumlah baris duplikat
df = df.drop_duplicates()
df['harga'] = df['harga'].astype(int) # memastikan tipe data harga adalah integer
print(df.dtypes)

5
tanggal      object
menu         object
kategori     object
harga         int64
terjual     float64
dtype: object


Awalnya terdapat 110 baris. Setelah proses dropna (hilang 3 baris jadi 107) dan drop_duplicates, jumlah baris sekarang menjadi [masukkan angka dari output df.shape milikmu].

In [ ]:
laris = df[df['terjual'] > 20] # filtering
urut = df.sort_values(by='terjual', ascending=False) # sorting
df['total_pendapatan'] = df['harga'] * df['terjual'] # kolom turunan
ringkasan = df.groupby('menu')['total_pendapatan'].sum() # agregasi
print(ringkasan)

menu
Es Jeruk       1460000.0
Es Teh         2144000.0
Kerupuk         972000.0
Mie Ayam       2410000.0
Nasi Goreng    3924000.0
Roti Bakar     1288000.0
Teh Hangat      459000.0
Name: total_pendapatan, dtype: float64


Lihat daftar nama makanan/minuman dan angka di sebelahnya. Cari angka yang paling besar. Contoh jawaban: "Berdasarkan hasil groupby, menu dengan total pendapatan tertinggi adalah [Sebutkan Nama Menunya, misal: Nasi Goreng] dengan total pendapatan sebesar [Sebutkan angkanya].

In [4]:
from google.colab import files

uploaded = files.upload()

Saving data_kantin.csv to data_kantin.csv


In [6]:
import pandas as pd

df = pd.read_csv("data_kantin.csv")

In [7]:
df.head()
df.info()
df.describe()
df.shape

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 110 entries, 0 to 109
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   tanggal   110 non-null    object 
 1   menu      107 non-null    object 
 2   kategori  110 non-null    object 
 3   harga     110 non-null    int64  
 4   terjual   104 non-null    float64
dtypes: float64(1), int64(1), object(3)
memory usage: 4.4+ KB


(110, 5)

In [8]:
df.isna().sum()          # cek missing value per kolom
df.duplicated().sum()    # cek jumlah baris duplikat

np.int64(5)

In [9]:
df = df.drop_duplicates().reset_index(drop=True)

In [10]:
df.groupby('harga')['menu'].unique()

,menu
harga,
2000,[Kerupuk]
3000,"[Teh Hangat, nan]"
4000,"[Es Teh, nan]"
5000,[Es Jeruk]
8000,[Roti Bakar]
10000,[Mie Ayam]
12000,"[Nasi Goreng, nan]"


In [11]:
harga_ke_menu = df.dropna(subset=['menu']).groupby('harga')['menu'].agg(lambda x: x.mode()[0])
mask_menu_kosong = df['menu'].isna()
df.loc[mask_menu_kosong, 'menu'] = df.loc[mask_menu_kosong, 'harga'].map(harga_ke_menu)

In [12]:
df['terjual'] = df['terjual'].fillna(df.groupby('menu')['terjual'].transform('median'))

In [13]:
df['tanggal'] = pd.to_datetime(df['tanggal'])   # dari teks jadi tanggal asli
df['terjual'] = df['terjual'].round().astype(int)  # dari desimal (hasil median) jadi bilangan bulat

In [14]:
df.isna().sum()       # harus 0 semua
df.duplicated().sum() # harus 0
df.dtypes             # cek semua tipe sudah benar

,0
tanggal,datetime64[ns]
menu,object
kategori,object
harga,int64
terjual,int64


In [15]:
print(df.isna().sum())
print(df.duplicated().sum())
print(df.dtypes)

tanggal     0
menu        0
kategori    0
harga       0
terjual     0
dtype: int64
0
tanggal     datetime64[ns]
menu                object
kategori            object
harga                int64
terjual              int64
dtype: object


In [16]:
df['pendapatan'] = df['harga'] * df['terjual']

In [17]:
penjualan_tinggi = df[df['terjual'] >= 25]

In [18]:
df_sorted = df.sort_values('pendapatan', ascending=False)

In [19]:
ringkasan_kategori = df.groupby('kategori').agg(
    total_terjual=('terjual', 'sum'),
    total_pendapatan=('pendapatan', 'sum')
).sort_values('total_pendapatan', ascending=False)

In [20]:
df.to_csv('dataset_bersih.csv', index=False)